### PR TITLE
Make `astyle` under `:ALEFix` operate on buffer

### PR DESCRIPTION
**Pull request by ankitpati**
Previously, `:ALEFix` made `astyle` operate on the disk file, and populated the buffer with the results, potentially losing work.

This PR regresses #3257, but that appears to be a bug with `astyle` itself, not with `ale`, so it should be fixed there anyways. I feel avoiding data-loss should be a higher goal than working around `astyle` bugs.

Thanks for your awesome work!

### CODE REVIEW

**Review by hsanson — COMMENTED** [review]
Change looks good to me but I do not use this tool so would be great if @jhlink  and @skorokithakis could test this change.

**Review by jhlink — CHANGES_REQUESTED** [review]
Tested, and see no change. 

Still, I'm against this change. Without any data *proving* that astyle formats code with some loss into a buffer, this is an unneeded, unsolicited change. 

Further, if the bug is a problem with astyle, why does ALE need to accommodate? 
Regressing support because a dependency has a bug makes little sense. 

If data loss is a concern, use another tool like clang. I will wait for astyler maintainers to release a fix.

**Comment by ankitpati** [comment]
@jhlink Steps to reproduce:

1. Ensure `astyle` is the only ALE fixer defined for C.
2. Open a `.c` file with valid C code.
3. Add a line of valid C code anywhere in the file.
4. Run `:ALEFix`
5. Observe that the added line has vanished.

**Comment by hsanson** [comment]
Took some time to re-read the related discussions and test this PR and unfortunately applying this PR reverts the initial issue.  If we do not use the --stdin flag then astyle fails to find and use any .astylerc or _astylerc configuration files within the project.   This would be a breaking change so I cannot approve this PR unless alternative solution for the .astylerc/_astylerc file issue is also included.

**Comment by jhlink** [comment]
Thanks for helping with the analysis @hsanson !

**Pull request closed without merging**
